### PR TITLE
Snhut 2386 edit user api

### PR DESCRIPTION
**Pull request by pran-sada**
### Description

- Edit user API integration.

- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-2386)

#### Checklist

- [ ] **Assumptions of User Story met (including acceptance criteria)**
- [ ] **All new necessary Unit tests were CREATED** to promote code coverage
- [ ] **All required unit tests are PASSING**
- [ ] The proposed functionality was tested manually locally, and it's working fine.
- [ ] PR was properly reviewed by a team member (at least 1 approval required to merge!)
- [ ] Unnecessary console.logs AND/OR comments were removed
- [ ] Feature ok-ed by UX designer (If needed - not for back end components) and Product Owner
- [ ] Ensure configuration and/or build changes documented

### CODE REVIEW

File: apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  });
+
+  it('should populate the userForm and open the modal', () => {
+    const spy = jest.spyOn(component.editModal, 'open')
+    component.ngOnChanges(
+      {
+        user: new SimpleChange(null, completedInputs, true)
+      }
+    );
+    expect(component.userForm.controls.firstName.value).toBe('Bob');
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts:51]: Lines 51-53 can be combined into a single expect statement,

File: apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  })
+
+  it('should not populate the userForm and open the modal', () => {
+    const spy = jest.spyOn(component.editModal, 'open');
+    component.ngOnChanges(
+      {
+        user: new SimpleChange(null, null, true)
+      }
+    );
+    expect(component.userForm.controls.firstName.value).toBe('');
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts:64]: Lines 64-67 can be combined into a single expect statement,

File: apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  })
+
+
+  it('should close the edit modal and reset the userForm', () => {
+    const spy = jest.spyOn(component.userChange, 'emit');
+   component.reviewing = null ;
+   component.userForm.setValue(completedInputs);
+   component.editModalHidden();
+   expect(spy).toBeCalledTimes(1)
+   expect(component.userForm.controls.firstName.value).not.toBeTruthy();
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts:77]: Lines 77-79 can be combined into a single expect statement,

**Comment by sada-ajith** [comment]
Please add unit test cases related to form validation

**Review by sada-ajith — CHANGES_REQUESTED** [review]
Please see my observations and fix it

File: apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts (hunk 1 above)
Comment (pran-sada) [reply apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts]: Combined into single expect statement

File: apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts (hunk 2 above)
Comment (pran-sada) [reply apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts]: combined into a single expect statement

File: apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts (hunk 3 above)
Comment (pran-sada) [reply apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.spec.ts]: combined into a single expect statement

**Review by sada-ajith — APPROVED** [review]
approved

**Pull request merged**